PROTO_tests/tests/dataframe_test.py

```diff
@@ -0,0 +1,704 @@
+        ord = df.sort_values()
+        assert_frame_equal(pd.DataFrame(data=userid, columns=["userID"]), ord.to_pandas())
+        ord = df.sort_values(ascending=False)
+        userid.reverse()
+        assert_frame_equal(pd.DataFrame(data=userid, columns=["userID"]), ord.to_pandas())
+
+        df = self.build_ak_df()
+        ord = df.sort_values(by="userID")
+        ref_df = self.build_pd_df()
+        ref_df = ref_df.sort_values(by="userID").reset_index(drop=True)
+        assert_frame_equal(ref_df, ord.to_pandas())
+
+        ord = df.sort_values(by=["userID", "day"])
+        ref_df = ref_df.sort_values(by=["userID", "day"]).reset_index(drop=True)
+        assert_frame_equal(ref_df, ord.to_pandas())
+
+        with pytest.raises(TypeError):
+            df.sort_values(by=1)
+
+    def test_intx(self):
+        username = ak.array(["Alice", "Bob", "Alice", "Carol", "Bob", "Alice"])
+        userid = ak.array([111, 222, 111, 333, 222, 111])
+        df_1 = ak.DataFrame({"user_name": username, "user_id": userid})
+
+        username = ak.array(["Bob", "Alice"])
+        userid = ak.array([222, 445])
+        df_2 = ak.DataFrame({"user_name": username, "user_id": userid})
+
+        rows = ak.intx(df_1, df_2)
+        assert rows.to_list() == [False, True, False, False, True, False]
+
+        df_3 = ak.DataFrame({"user_name": username, "user_number": userid})
+        with pytest.raises(ValueError):
+            rows = ak.intx(df_1, df_3)
+
+    def test_apply_perm(self):
+        df = self.build_ak_df()
+        ref_df = self.build_pd_df()
+
+        ord = df.sort_values(by="userID")
+        perm_list = [0, 3, 1, 5, 4, 2]
+        default_perm = ak.array(perm_list)
+        ord.apply_permutation(default_perm)
+
+        ord_ref = ref_df.sort_values(by="userID").reset_index(drop=True)
+        ord_ref = ord_ref.reindex(perm_list).reset_index(drop=True)
+        assert_frame_equal(ord_ref, ord.to_pandas())
+
+    def test_filter_by_range(self):
+        userid = ak.array([111, 222, 111, 333, 222, 111])
+        amount = ak.array([0, 1, 1, 2, 3, 15])
+        df = ak.DataFrame({"userID": userid, "amount": amount})
+
+        filtered = df.filter_by_range(keys=["userID"], low=1, high=2)
+        assert filtered.to_list() == [False, True, False, True, True, False]
+
+    def test_copy(self):
+        username = ak.array(["Alice", "Bob", "Alice", "Carol", "Bob", "Alice"])
+        userid = ak.array([111, 222, 111, 333, 222, 111])
+        df = ak.DataFrame({"userName": username, "userID": userid})
+
+        df_copy = df.copy(deep=True)
+        assert_frame_equal(df.to_pandas(), df_copy.to_pandas())
+
+        df_copy.__setitem__("userID", ak.array([1, 2, 1, 3, 2, 1]))
+        assert df["userID"].to_list() != df_copy["userID"].to_list()
+
+        df_copy = df.copy(deep=False)
+        df_copy.__setitem__("userID", ak.array([1, 2, 1, 3, 2, 1]))
+        assert_frame_equal(df.to_pandas(), df_copy.to_pandas())
+
```

Comment:
I noticed `test_save` is also dropped. I should absolutely add this in my PR #2586 and try to find other spots that might be doing their own IO testing. This is an action item for me, nothing required on your part

Reply:
That was my intention as well. Thanks.